src/averageAges.js

```diff
@@ -37,7 +40,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let answer;
+
+  if (withChildren) {
+    const mothersNames = people
+      .filter((name) => name.mother !== null)
+      .map((name) => name.mother);
+
+    answer = people.filter(obj => mothersNames.indexOf(obj.name) !== -1);
+  } else {
+    answer = people.filter(obj => obj.sex === 'f');
+  }
+
+  return answer.reduce((sum, human) =>
+    sum + (human.died - human.born), 0) / answer.length;
```

Comment:
it would be great to use ternary here 
```
const women = withChildren 
? 
:
```

Reply:
If there are no second param u should only find women.
if there are, you need to check if she has a child

Reply:
did as much as I understood)